### ISSUE TITLE
Extension is not compatible with new (Oct 2017) Google Calendar

### ISSUE THREAD

**Issue by tomviner** [tomviner/gcal-unlimited-weeks#5]
Google released a [new look Calendar](https://techcrunch.com/2017/10/17/google-calendar-on-the-web-gets-a-fresh-new-look/) this week. The code is no longer compatible for at least these reasons:

- the buttons are replaced with a dropdown
- the html appears to have changed therefore the selectors no longer work

![image](https://user-images.githubusercontent.com/167319/31792860-576566ba-b515-11e7-9c61-4ddd8886ebca.png)

Need to check the underlying method, explained in the README will still work.

**Comment by lucasverra** [comment]
please let us know when compatible with new calendar look so we can upgrade !

**Comment by wghudson** [comment]
Agreed, would really like to get this functionality back in the new calendar. 
I put off using the new interface as long as I could just to keep using this extension.
The new calendar does have 'custom views', but the longest period is 4 weeks and this requires descending into a menu to change it... ew.

**Comment by davelab6** [comment]
Would love to have this :)

**Comment by fservant** [comment]
Please, do update your plugin to the new calendar. It's tremendously useful, and there is nothing like it out there. I'd be willing to help out.

**Comment by strepto42** [comment]
Would love this too... now that google have forced the new calendar on everyone...

**Comment by tomviner** [comment]
I appreciate the interest in this project. Here's an update.

## Problem

I'm still searching for a way to show arbitrary numbers of weeks on the new Google Calendar interface.

The approach I implemented for the old interface was a two part manoeuvre: click next month repeatedly during a click drag over the mini calendar. The click drag would lift up on a new month than it started on. (It's Javascript not a real mouse button, so we do these things at the same time). As far as I can see this approach doesn't work with the new interface.

I discovered this by dragging my mouse over the mini calendar, using the setTimeout code below to click on the next month button, then lifting my mouse button on the new month. This should allow showing like 8 weeks, but the interface would only select a maximum of 4 weeks.

```javascript
// Run this in the Chrome Dev Tools console
// Then in five seconds the next month button will be clicked.
var next_month_button_path = '#drawerMiniMonthNavigator > div > div > div > content > span > span';
var next_month_button = document.querySelectorAll(next_month_button_path)[1];
var mouse_click = new MouseEvent('click', {bubbles: true});
setTimeout(() => next_month_button.dispatchEvent(mouse_click), 5000);
```

## Ideas

The Google Calendar Javascript knows how to show more than 4 weeks, as it shows 5 in the month view. Perhaps there's a function somewhere we could call to show other ranges?

Help appreciated! cc @fservant :-)

**Comment by msbarrios** [comment]
Yeah, even if you select 5 weeks in the minical you will still only get a max of 4. :( But it is possible to show more as you said. I was able to show 8, 12 weeks... probably any number I want. But it involves adding a breakpoint in the javascript and modifying a string. It's actually simple.... once you find the point for the breakpoint. And it's really stable, clicking on "previous period"/"next period" maintains the selection of weeks. But I couldn't find how to get to that function by user interaction/request, for the life of me.

Here's what I did: find in the javascript sources the string "a.duration", add a breakpoint at the line "a=HT(IT(this.V.H(),a.duration.Oa),a.start.Ta()).H();" (variables and function names will vary (HT, IT, V, H, Ta, Oa) ). After that, select 2 weeks in the minical. It will stop at the breakpoint. Then, execute only that line, "a" will have a value similar to:
"./r/custom/2/w/2018/2/24"
This being: "./r/custom/[weeks selected]/w/[first day of first week selected]"
So in this case, I had selected from 2018-02-24 to 2018-03-09. Still paused on that line, go to the console and change the value of "a" to have any number of weeks you want. For example:
a = "./r/custom/6/w/2018/2/24";
Hit enter, tell the browser debugger to proceed with execution. Voila, you see 6 weeks starting on 2018-02-24.
The javascript for this new calendar is crazy, I'm sure they compile the source through some obfuscator. :(

**Comment by lucasverra** [comment]
@msbarrios Is this a one time "forever" solution or to be repeated at each calendar loading ?

**Comment by msbarrios** [comment]
To be repeated at each loading. :(
I wasn't proposing as a solution for "regular users", I was hoping it would help @tomviner or some other extension developer to find a way to create a new extension for the new google calendar. It's too much work to do every time you want to use the calendar.

**Comment by C00woot** [comment]
Dear Tom, I miss this functionality so much, that I would be willing to PAY you to re-release this app for the latest version of Calendar.  Please consider this, it would be incredible.

**Comment by fservant** [comment]
@msbarrios -- I can't seem to make your workaround work. Would you be willing to make a quick screencast on how you did it? It would be really useful for me (and I'm sure many others) even if it has to be repeated at each calendar loading. Thank you so much in advance!

**Comment by C00woot** [comment]
@msbarrios @fservant -- I too would be very grateful to get some guidance on how this could work please.

I honestly have so much need to get this functionality back. I would be happy to help finance a solution from a developer if someone is up to the task...

**Comment by akinariobi** [comment]
Can I contribute somehow to this project?
P.S. still waiting for re-release of this app for the latest version of Calendar.

**Comment by msbarrios** [comment]
Hi guys, sorry, the workaround doesn't work anymore anyway, it's been a few months. And they changed everything again. :(
